woocommerce/woocommerce-ios
# [Woo POS] Remove POS access reliance on beta feature toggle switch

Ref: p1725337084101939-slack-C070SJRA8DP

https://github.com/woocommerce/woocommerce-ios/pull/13815 must be merged first.
Closes: #13838


## Description
This PR removes the toggle switch from `Menu > Experimental Features > Point of Sale`, so rendering the point of entry to access POS only relies on the `POSEligibilityChecker.isEligible` publisher now, rather than both the toggle + the publisher. The POS row will only appear on developer builds, as long as the store is eligible for POS.

There are no changes in `isEligible`, since this already checks for `displayPointOfSaleToggle`, which is set to only return true on `buildConfig == .localDeveloper || buildConfig == .alpha` 

## Changes
* Removes reliance on the beta feature flag toggle and related AppSettingsAction / AppSettingsStore code
* Updates feature flag name

## Testing:
1. On an eligible store for POS (WooCommerce 6.6+, IPP onboarding completed, whitelisted in `mobile/feature-flags/`, and using an iPad):
2. Run the app in `Debug` build configuration:
<img width="974" alt="Screenshot 2024-09-04 at 13 19 06" src="https://github.com/user-attachments/assets/ac797506-6e15-4b1a-b128-68c78f7f02e3">

3. Observe there is no toggle under `Menu > Experimental Features > Point of Sale`
4. Observe that the POS row is visible

![Simulator Screenshot - iPad Air 11 - iOS 17 5 M2 - 2024-09-04 at 14 19 21](https://github.com/user-attachments/assets/30a873e8-7bb2-4a4f-b6c5-3f7393aa5bde)

5. Repeat for `Release` build configuration
6. Observe there still is no toggle under `Menu > Experimental Features > Point of Sale`
7. Observe that the POS row is no longer visible

![Simulator Screenshot - iPad Air 11 - iOS 17 5 - 2024-09-04 at 14 26 26](https://github.com/user-attachments/assets/a404290b-b946-4c11-bcca-2914a3f70985)

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
